wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/compose/deserializer/EnvironmentDeserializer.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.environment.server.compose.deserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+/**
+ * Deserializer of environment section of Docker compose file.
+ *
+ * @author Dmytro Nochevnov
+ */
+public class EnvironmentDeserializer extends JsonDeserializer<Map<String, String>> {
+
+    public static final String ARRAY_ITEM_DIVIDER        = "=";
+    public static final String UNSUPPORTED_VALUE_MESSAGE = "Unsupported value '%s'.";
+
+    @Override
+    public Map<String, String> deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+        Object environment = jsonParser.readValueAs(Object.class);
+
+        /* Parse dictionary in view of:
+           "environment:
+             key1: value1
+             key2: value2"
+        */
+        if (environment instanceof Map) {
+            return (Map<String, String>) environment;
+        }
+
+        /* Parse array in view of:
+           "environment:
+             - key1=value1
+             - key2=value2"
+        */
+        try {
+            if (environment instanceof List) {
+                // convert array to Map<String, String>{ key1: value1, key2: value2 }
+                Map<String, String> map = new HashMap<>();
+                for (String item : (List<String>) environment) {
+                    map.putAll(parseEnvironmentItem(item));
```

Comment:
I'd make private method `Map<String, String> parseEnvironmentItem(List<String> items)` because it is quite strange to  invoke method putAll when you return one and only one element. Or you can return Pair. I prefer first way